src/java.base/unix/native/libnio/ch/FileChannelImpl.c

```diff
@@ -52,11 +53,21 @@
 
 static jfieldID chan_fd;        /* jobject 'fd' in sun.nio.ch.FileChannelImpl */
 
+#if defined(__linux__)
+typedef ssize_t copy_file_range_func(int, loff_t*, int, loff_t*, size_t,
+                                     unsigned int);
+static copy_file_range_func* my_copy_file_range_func = NULL;
+#endif
+
 JNIEXPORT jlong JNICALL
 Java_sun_nio_ch_FileChannelImpl_initIDs(JNIEnv *env, jclass clazz)
 {
     jlong pageSize = sysconf(_SC_PAGESIZE);
     chan_fd = (*env)->GetFieldID(env, clazz, "fd", "Ljava/io/FileDescriptor;");
```

Comment:
At some point we need to remove chan_fd, instead map0 should provide the fd.

Reply:
I do not understand exactly what is intended here.

Reply:
I'd like to see chan_fd removed. To do that would require changing map0 to take a file descriptor. For another PR of course.

Reply:
https://bugs.openjdk.java.net/browse/JDK-8286671